server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -1119,6 +1120,14 @@ private void registerQueryParsers(List<SearchPlugin> plugins) {
             )
         );
 
+        registerQuery(
+            new QuerySpec<>(
+                RerankQueryBuilder.NAME,
+                RerankQueryBuilder::new,
+                parser -> { throw new IllegalArgumentException("[rerank] queries cannot be provided directly"); }
+            )
+        );
+
```

Comment:
While this is a clever idea, I don't think this is the abstraction we want. It may be that on the individual node itself, the reranker provides something like this. But to me we should be passing around a `RerankerContext` object or something that tracks stuff instead of hijacking query.

Reply:
Changed this to pass the additional queries along as part of the `RankBuilder` using the internal transport protocol.